From 9948d7673433bee8505d7240c172deb0e03d0163 Mon Sep 17 00:00:00 2001
From: Wilson Ding <dingwei@marvell.com>
Date: Wed, 2 Mar 2016 12:28:28 +0800
Subject: [PATCH 0921/1240] fix: makefile: restore the doimage for armada3700
 and armada38x

- The patch fixed the incorrect location of adding doimage
  target.
- It should be placed before "all:" target which
  is dependent on "all-y".
- It also changes to a better manner to add or remove the
  platform to enable doimage tool. If new platform relies
  on the doimage target, it needs to add new branch. If
  one platform no longer use doimage tool, just simply
  remove the existing branch. For A38X, it always uses
  doimage tools to generate the boot image.

Change-Id: I50aaaf9f27166a5e05813f94cfb49e8fa2358fd5
Signed-off-by: Wilson Ding <dingwei@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27981
Reviewed-by: Hanna Hawa <hannah@marvell.com>
Tested-by: Hanna Hawa <hannah@marvell.com>
---
 Makefile | 11 +++++++----
 1 file changed, 7 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index d4affc5..6da47b7 100644
--- a/Makefile
+++ b/Makefile
@@ -768,6 +768,13 @@ append = cat $(filter-out $< $(PHONY), $^) >> $@
 quiet_cmd_pad_cat = CAT     $@
 cmd_pad_cat = $(cmd_objcopy) && $(append) || rm -f $@
 
+# Add doimage target for mvebu soc
+ifeq ($(CONFIG_TARGET_ARMADA_38X), y)
+ALL-$(CONFIG_MVEBU) += doimage
+else ifeq ($(CONFIG_TARGET_ARMADA_3700), y)
+ALL-$(CONFIG_MVEBU) += doimage
+endif
+
 all:		$(ALL-y)
 ifneq ($(CONFIG_SYS_GENERIC_BOARD),y)
 	@echo "===================== WARNING ======================"
@@ -892,8 +899,6 @@ DOIMAGE_LIBS_CHECK = \
 		echo "DOIMAGE=$(DOIMAGE)" >&1; \
 	fi
 
-ALL-$(CONFIG_MVEBU) += doimage
-
 # Start with creation of UART images:
 # - Create TIM descriptor with UART signature
 # - Create binary TIM and UART downloadable images (*_h.*)
@@ -958,8 +963,6 @@ MKIMAGEFLAGS_u-boot.kwb = -n $(srctree)/$(CONFIG_SYS_KWD_CONFIG:"%"=%) \
 MKIMAGEFLAGS_u-boot.pbl = -n $(srctree)/$(CONFIG_SYS_FSL_PBL_RCW:"%"=%) \
 		-R $(srctree)/$(CONFIG_SYS_FSL_PBL_PBI:"%"=%) -T pblimage
 
-ALL-$(CONFIG_MVEBU) += doimage
-
 u-boot.img u-boot.kwb u-boot.pbl: u-boot.bin FORCE
 	$(call if_changed,mkimage)
 
-- 
1.9.1

